### PR TITLE
Force usage of the system console when initializing the proactive-cli

### PR DESCRIPTION
**Pull request by paraita**
This is intended to avoid JLine errors if the proactive-client command is sent to the background (e.g `nohup ./proactive-client -lj &`)
Minor refactor over the naming of the SYSTEM console

### CODE REVIEW

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java [hunk 1]
```diff
@@ -46,6 +53,17 @@ public AbstractIModeCommand() {
...
     public void execute(ApplicationContext currentContext) throws CLIException {
+        CommandFactory commandFactory = CommandFactory.getCommandFactory(CommandFactory.Type.ALL);
+        AbstractDevice console = null;
+        try {
+            console = AbstractDevice.getConsole(AbstractDevice.JLINE);
+            ((JLineDevice) console).setCommands(ObjectArrays.concat(commandFactory.supportedCommandEntries(),
+                                                                    CommandSet.INTERACTIVE_COMMANDS,
+                                                                    CommandSet.Entry.class));
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (sonarqube-activeeon) [inline rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java]: MINOR Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [EntryPoint.java#L179](https://github.com/ow2-proactive/scheduling/blob/3b3c1c98ce164f8a163ab82d15537be7333ab1b3/rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/EntryPoint.java#L179): Use try-with-resources or close this "PrintWriter" in a "finally" clause. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java [hunk 2]
```diff
@@ -46,6 +53,17 @@ public AbstractIModeCommand() {
...
     public void execute(ApplicationContext currentContext) throws CLIException {
+        CommandFactory commandFactory = CommandFactory.getCommandFactory(CommandFactory.Type.ALL);
+        AbstractDevice console = null;
+        try {
+            console = AbstractDevice.getConsole(AbstractDevice.JLINE);
+            ((JLineDevice) console).setCommands(ObjectArrays.concat(commandFactory.supportedCommandEntries(),
+                                                                    CommandSet.INTERACTIVE_COMMANDS,
+                                                                    CommandSet.Entry.class));
+        } catch (IOException e) {
+            handleError(String.format("An error occurred while initializing the console:"), e, currentContext);
```
Comment (sonarqube-activeeon) [inline rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java]: MAJOR String contains no format specifiers. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/EntryPoint.java [hunk 3]
```diff
@@ -184,22 +176,24 @@ private void executeCommandList(List<Command> commandList, ApplicationContext cu
...
-            if (isDebugModeEnabled) {
-                if (cause instanceof CLIException && ((CLIException) cause).stackTrace() != null) {
-                    writer.printf("%nStack trace: %s%n", ((CLIException) cause).stackTrace());
+            if (cause != null) {
+                if (cause.getMessage() != null) {
+                    writer.printf("%n%nError message: %s%n", cause.getMessage());
+                }
+
+                if (isDebugModeEnabled) {
+                    if (cause instanceof CLIException && ((CLIException) cause).stackTrace() != null) {
```
[the remark this answers is not in the document]
Comment (paraita) [reply rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/EntryPoint.java]: I'll add comments and clarify what this does, and merge my changes thanks @fviale for the review !

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java [hunk 4]
```diff
@@ -46,6 +53,17 @@ public AbstractIModeCommand() {
 
     @Override
     public void execute(ApplicationContext currentContext) throws CLIException {
+        CommandFactory commandFactory = CommandFactory.getCommandFactory(CommandFactory.Type.ALL);
+        AbstractDevice console = null;
+        try {
+            console = AbstractDevice.getConsole(AbstractDevice.JLINE);
```
Comment (fviale) [inline rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/AbstractIModeCommand.java]: similarly, if you can add comments

**Pull request merged**